packages/lit-dev-content/src/github/github-types.d.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+export type GitHubSigninReceiverMessage =
+  | {code: string; error?: undefined}
+  | {code?: undefined; error: string};
```

Comment:
Would you ever have both `code` and `error`?

```typescript
export type GitHubSigninReceiverMessage = {code: string} | {error: string};
```

Reply:
Nope, can't have both. But turns out it works better to include the `error?: undefined` cases in the type because otherwise when we check for the `error` property, TypeScript will say "Hey, this is a union where one of the things doesn't have an error property, that's not safe!".

https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEYD2A7AzgF3lAXPA3sqHpjAJYoDmAvvAD4EgwxIwkblXUDcAUGQDN4ACgCEUAHRMWMAJQFe8RKjRIIICRCSVhkoiFl9qvXqEiwEydFgBGORQX3tOlADTxprAPx4ArilABChBgWjoHQiRQH3h-QODgd082eFIKGj5BERspZlZ5fAcrVXVNbWEc-UNeaiA

(Also TIL you can comment on multiple lines!)